--- cpplint.py.orig	2013-07-12 12:57:05.000000000 -0700
+++ cpplint.py	2013-09-01 14:11:42.995802695 -0700
@@ -3909,7 +3909,7 @@
   # When reading from stdin, the extension is unknown, so no cpplint tests
   # should rely on the extension.
   if (filename != '-' and file_extension != 'cc' and file_extension != 'h'
-      and file_extension != 'cpp'):
+      and file_extension != 'cpp' and file_extension != 'c'):
     sys.stderr.write('Ignoring %s; not a .cc or .h file\n' % filename)
   else:
     ProcessFileData(filename, file_extension, lines, Error,
